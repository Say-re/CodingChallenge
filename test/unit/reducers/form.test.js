import formReducer, { defaultState } from '../../../src/reducers/form';

describe('Form Reducer Test', () => {
  test('Match Snapshot', async () => {
    await expect(defaultState).toMatchSnapshot()
  });
  test('HANDLE_INPUT_CHANGE', async () => {
    const responseState = formReducer(defaultState, {
      type: 'HANDLE_INPUT_CHANGE',
      updateValue: 'Bob',
      updateField: 'firstName'
    })
    await expect(responseState.firstName).toEqual('Bob');
  });
})