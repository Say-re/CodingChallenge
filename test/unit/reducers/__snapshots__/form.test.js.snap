// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Reducer Test Match Snapshot 1`] = `
Object {
  "firstName": "Travis",
  "lastName": "Rose",
  "userAddress": "123 Candycane Lane",
  "userCity": "North Pole",
  "userEmail": "annonymous@gmail.com",
  "userExperience": "Less than 1 year",
  "userJob": "Manager",
  "userShortAnswer": "",
  "userState": "Ice Caps",
  "userTechnologyExperience": Array [
    "css",
  ],
  "userTelephone": "808-029-0394",
  "userTest": "No",
  "userZipcode": "123987",
}
`;
